### PR TITLE
provider status route for monitoring

### LINKED ISSUE #397: Notification provider report program down
After we implement new provider report program make sure we have method to track status of program and get  notification if program cannot send report to chain.

Notification doesn't need to start with each program. You can just watch all request and check report not sent by validator program (Check result)

Or have service to periodically check status of program directly(health check)

### PR DESCRIPTION
**Pull request by evilpeach**
Fixes #397 
[Test on](http://rpc.alpha-debug.d3n.xyz:1317/d3n/provider_status)

### CODE REVIEW

File: chain/client/client.go [hunk 1]
```diff
@@ -129,9 +131,51 @@ func GetHealthStatus(cliCtx context.CLIContext) http.HandlerFunc {
...
+
+		block, err := cliCtx.Client.Block(nil)
+		if err != nil {
+			rest.WriteErrorResponse(w, http.StatusInternalServerError, err.Error())
+			return
+		}
+
+		numReporters := len(request.Reports)
+		status := "GOOD"
+		// TODO: Hard code provider count threshold
```
Comment (sorawit) [inline chain/client/client.go]: TODO: Remove hard-coded ...

**Review by sorawit — CHANGES_REQUESTED** [review]

File: chain/client/client.go [hunk 2]
```diff
@@ -129,9 +131,51 @@ func GetHealthStatus(cliCtx context.CLIContext) http.HandlerFunc {
...
+		if uint64(block.Block.Height) > request.TargetBlock && numReporters < 3 {
+			fmt.Printf(`BAD ------- requestId: %s, reports: %d`, requestId, numReporters)
+			status = "BAD"
+		}
+
+		w.Write([]byte(fmt.Sprintf(`height: %d, requestId: %s, reports: %d, status: %s`, block.Block.Height, requestId, numReporters, status)))
```
Comment (taobun) [inline chain/client/client.go]: Return in json format will be better raw string

**Review by taobun — CHANGES_REQUESTED** [review]

File: chain/client/client.go [hunk 3]
```diff
@@ -129,9 +131,61 @@ func GetHealthStatus(cliCtx context.CLIContext) http.HandlerFunc {
...
+		if err != nil {
+			rest.WriteErrorResponse(w, http.StatusInternalServerError, err.Error())
+			return
+		}
+
+		numReporters := len(request.Reports)
+		status := "GOOD"
+		// TODO: Remove hard-coded provider count threshold
+		if uint64(block.Block.Height) > request.TargetBlock && numReporters < 3 {
+			fmt.Printf(`BAD ------- requestId: %s, reports: %d`, requestId, numReporters)
```
Comment (sorawit) [inline chain/client/client.go:167]: Is this intended to stick?

**Review by sorawit — CHANGES_REQUESTED** [review]

File: chain/client/client.go (hunk 3 above)
Comment (evilpeach) [reply chain/client/client.go:167]: Yes

**Review by taobun — APPROVED** [review]

**Review by sorawit — APPROVED** [review]

**Pull request merged**